argonaut/src/test/scala/pathy/argonaut/PosixCodecJsonSpec.scala

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2014 - 2015 SlamData Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pathy.argonaut
+
+import pathy.Path, Path._
+import pathy.scalacheck.PathyArbitrary._
+
+import argonaut._
+import org.specs2.mutable
+import org.specs2.ScalaCheck
+
+final class PosixCodecJsonSpec extends mutable.Specification with ScalaCheck {
+  import PosixCodecJson._
+
+  "RelFile Codec" ! prop { rf: RelFile[Sandboxed] =>
+    CodecJson.codecLaw(CodecJson.derived[RelFile[Unsandboxed]])(unsandbox(rf))
+  }
+
+  "RelDir Codec" ! prop { rd: RelDir[Sandboxed] =>
+    CodecJson.codecLaw(CodecJson.derived[RelDir[Unsandboxed]])(unsandbox(rd))
+  }
+
+  "AbsFile Codec" ! prop { af: AbsFile[Sandboxed] =>
+    CodecJson.codecLaw(CodecJson.derived[AbsFile[Sandboxed]])(af)
+  }
+
+  "AbsDir Codec" ! prop { ad: AbsDir[Sandboxed] =>
+    CodecJson.codecLaw(CodecJson.derived[AbsDir[Sandboxed]])(ad)
+  }
+}
```

Comment:
I was only giving an example. The point is that these tests (obviously) verify only one thing, the codec law (`encode >>> decode == id`), and surely that's not the only important behavior. Since you're only feeding the result of `encode` to `decode`, you have no verification of what happens when `decode` gets any other input. And since you never look at the encoded form, you have no verification that it is what you expect or that it doesn't change when the code changes.

Many of those inputs should result in decode errors. How are errors captured? Is there any input that makes the decoder fail in the wrong way (throws)? 

Are there any JSON values that can't be produced by `encode`, but which should be handled by `decode`?

What is the actual encoded form of any particular path? What if we change our minds on how `posixCodec` handles some case—do we want that to be reflected in JSON or not?